### ISSUE TITLE
Dataforms: Quick Edit controls labels are not accessible or empty

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68746]
### Description

1
In the data views quick edit, the controls to edit the edited object properties use an aria-label that completley mismatches the visible text. This pattern isn't accesisble and must be avoided. For example:

- Vieibls text: Published
- aria-label="Edit Status & Visibility"

or, another example:

- Visible text: Closed
- aria-label="Edit Discussion"

This pattern was already reported as not accessible almost 8 years ago in https://github.com/WordPress/gutenberg/issues/470 and htere's no full solution yet. In the post summary panel, it as only been mitigated but still the visible text and the aria-label mismatch. See https://github.com/WordPress/gutenberg/issues/63992
Buttons should be labeled to communicate what they do. For no reason a value or state should be used to bale interactive controls.

2
Multiple selection.
When more than one object is selected, the Quick edit panel shows a reduced set of properties that can be edited. Currently:
- Author
- Date
- Discussion

![Image](https://github.com/user-attachments/assets/e48234c1-c7c2-4b6c-9cd3-1aef46b74103)

The Author ane Discussion buttons are empty. The Discussion button is not visible at all because, well, it's empty.
The Date button shows the current date / time.
Still, these buttons use aria-labels that completely mismach the visible (or missing) text:
- aria-label="Edit Author"
- aria-label="Edit Date"
- aria-label="Edit Discussion"

These aria-labels also deviate from the pattern used in the Post summary where, at least, the aria-label is a combination of the action and value. Though slightly better, that pattern isn't accessible as well as reported on https://github.com/WordPress/gutenberg/issues/63992 and originally on https://github.com/WordPress/gutenberg/issues/470.
- aria-label="Change author: admin"
- aria-label="Change date: December 16, 2024 11:38 am UTC+0"
- aria-label="Change discussion options"

it is worth reminding that the Classic admin 'Bulk edit' shows a 'No change' label for properties of multiple selected obects. Screenshot:

![Image](https://github.com/user-attachments/assets/0f049603-e0ba-447f-92fe-e3e9e24b908d)

### Step-by-step reproduction instructions

- Go to the WP admin > Gutenberg > Experiments and enable 'Data Views: add Quick Edit'.
- Go to the Site editor > Pages and switch to the Table view for making testing easier.
- Click the 'Details' icon button at the top of the page to open the Quick Edit panel.
- Select one page.
- Observe the visible text of all the buttons in the Quick Edit panel mismatch the buttons aria-label attributes.
- Select more than one page.
- Observe the Author button only shows an icon.
- Observe the Discussion button is empty and not visible (unless focused).

## Expected behaviour

- The visible text to not mismatch the aria-label or the aria-label to be entirely removed and the buttons to use a meaningfultext.
- The buttons to not be empty.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by n2erjo00** [comment]
@afercia What you think should that HTML use table tag instead of bunch divs. I think it would make semantically more sense and eliminate need to use aria-label(s) to provide context

```
<table>
 <tr>
  <th scope="row">Status & Visibility</th>
  <td>
   <button>Draft</button>
  <td>
 </tr>
  …
</table>
```

Second issue to me seems simple to me just add "No change" (or something more positive sounding 😄 ) as placeholder.

**Comment by talldan** [comment]
Just updating the issue title as I believe it's the Dataforms component that's used for quick editing.

**Comment by ntsekouras** [comment]
👋 - @afercia how this issue translates with the current status of DataForms and quick edit, where we have the dedicated `edit` (pencil) button?
